### DISCUSSION TITLE
Welcome to Ivy-Octernships-ML Discussions!

### DISCUSSION THREAD

**Discussion by ashwinexe** [GitHub-Octernships/Ivy-Octernships-ML#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to answer and connect with all Octernships related conversations. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Sanchay-T** [comment]
"Hey Guys!

This is my first time participating in an Octernship and I'm eager to learn and contribute.

I have already applied and have the project on my dashboard. I understand that my task is to choose and implement one function from the list provided. However, I'm still somewhat unclear about the exact steps I should take after selecting a function.

I understood that I should comment on the corresponding issue to let others know that I'm working on it, set up my local environment, familiarize myself with Ivy, write the function using Ivy, and then write tests for it. After these steps, I should commit my changes, push to my forked repository, and finally create a pull request to the original repository.

While I understand these steps in theory, I am not sure about the exact details and practical implications. For example, how do I ensure that my function implementation fits within Ivy's guidelines? Are there any specific standards or templates for writing tests for the function?

This is my first major open-source contribution, so I'd appreciate any additional guidance, examples, or resources to help me navigate this process. I am looking forward to learning from all of you and contributing to this exciting project. Thank you!"

**Reply by AlexandraPosta** [reply]
Hello Sanchay! 😊

We are really happy to hear you’re interested in working with us at Ivy! You have understood the task really well, well done, all of the steps mentioned are correct👏. 

Your code should look similar to our current [frontend function implementations](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/tree/master/ivy/functional/frontends/paddle) and [frontend tests](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/tree/master/ivy_tests/test_ivy/test_frontends/test_paddle). There are a lot of examples written by other members of our community, so feel free to check their contributions to get a sense of what you are meant to do. For example, you can view all of the current [pull requests](https://github.com/unifyai/ivy/pulls?page=1&q=is%3Apr+is%3Aopen+paddle+label%3A%22Paddle+Frontend%22). 

Additionally, the contribution process is explained in our [docs](https://unify.ai/docs/ivy/) and also in this [video ](https://www.youtube.com/live/Zdkr9eQfP5k?feature=share)(around minute 23:00). 

Happy coding! 🎉👨‍💻

**Reply by Sanchay-T** [reply]
I completed the task and created a pr , but I know something is wrong. Can anyone help me , or how do I ask for help for that it. Its
#21251

**Reply by ashwinexe** [reply]
@Sanchay-T A valid submission is making a PR on your Octernships Assignment repository. 
For you it'll be https://github.com/GitHub-Octernships/ivy-octernships-ml-Sanchay-T

Please merge your PR before deadline as mentioned on the submission steps.

**Reply by Sanchay-T** [reply]
But in the octernship repo , it was said to create a pr there and then they will merge it right?I did that , can you guide me what to do after that?

**Reply by ashwinexe** [reply]
The PR should be created and merge in your own repository for review.

**Reply by Sanchay-T** [reply]
I am sorry , but I am having trouble understanding. This is what I have done according to the ivy documentation : [https://github.com/unifyai/ivy/pull/21303](url) , if that is the case , should I just create a pr for the same in [https://github.com/GitHub-Octernships/ivy-octernships-ml-Sanchay-T](url)? This is my first time contributing to open source.

**Reply by ashwinexe** [reply]
> should I just create a pr for the same in [https://github.com/GitHub-Octernships/ivy-octernships-ml-Sanchay-T](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/discussions/url)?

That is correct!

If this is your first time contributing to open source. I'd also recommend checking out some resources.
- https://opensource.guide/how-to-contribute/#finding-a-project-to-contribute-to
- https://dev.to/opensauced/how-open-source-helped-me-get-a-github-octernship-4f69

**Reply by sahil-sagwekar2652** [reply]
Thank you Ashwin for the guidance!

**Reply by SumantxD** [reply]
do we ask for a review ? or will they review after the deadline?

**Reply by rogn02** [reply]
So we don't merge it to master, we just keep the PR open.

**Reply by ashwinexe** [reply]
Please merge the PR to master before deadline so it can be reviewed. Your entire repository is reviewed instead of open PRs.

**Comment by sahil-sagwekar2652** [comment]
Hello everyone, I am Sahil. I am a final year student pursuing my Bachelor's degree in Information Technology

I'm new to the Ivy community, just joined in through Octernships. I was wondering if we have to claim the issues, if so then how should one go about it? Looking forward to exploring the project, the idea is really interesting!

**Comment by UdBe** [comment]
Hello,
My name is Uday, I am an undergraduate engineering student from India. I am passionate about Machine Learning and Data science applications. 
I have completed the task according to the Task Submission Section in the Ivy-Octernship-ML Repository. Can someone please guide me on the next steps.

Thanks!

**Reply by ashwinexe** [reply]
We will start reviews after the deadline of Assignment.

**Reply by UdBe** [reply]
> We will start reviews after the deadline of Assignment. 

Alright, Thanks 😀

**Comment by ManoBharathi93** [comment]
Hi @ashwinexe  I have completed the task and made a PR at unify/ivy repository..
MY PR: https://github.com/unifyai/ivy/pull/21419
Should I fill this field in  github-octernship repo ? 
<img width="581" alt="image" src="https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/88357044/5f2058cc-0bd6-4797-855f-a9555e233d87">
Whether I should make pr to github-octernship repo ? 
<img width="941" alt="image" src="https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/88357044/a6f9ea1f-e41f-4120-91c7-78a381c5676e">

**Reply by ashwinexe** [reply]
@ManoBharathi93 PR on https://github.com/unifyai/ivy may not be valid. Please create PRs in your own repository.

**Reply by ashwinexe** [reply]
Check this discussion to understand how submission works https://github.com/GitHub-Octernships/Ivy-Octernships-ML/discussions/4#discussioncomment-6665922

**Comment by rohinish404** [comment]
Can i usee github codespaces for setup instead of local installation?

**Reply by ashwinexe** [reply]
Sure

**Reply by sabre-code** [reply]
@ashwinexe Just wanted to confirm, I should be cloning the private repo with username right?

**Reply by sahil-sagwekar2652** [reply]
Yes @sabre-code, it should be https://github.com/GitHub-Octernships/ivy-octernships-ml-sabre-code for you

**Reply by sabre-code** [reply]
thanks 👍

**Comment by GargiMittal** [comment]
Hi! accidentally deleted my octernship repository is there anyway to recover it? I also can't reapply

**Reply by ashwinexe** [reply]
Hey @GargiMittal
I looked up and found this: https://github.com/GitHub-Octernships/ivy-octernships-ml-GargiMittal

**Reply by GargiMittal** [reply]
Shows " page not found"  for me

**Reply by ashwinexe** [reply]
Added you back.

**Reply by GargiMittal** [reply]
Thank You So Much!!! This is my first time trying to contribute in a big project, All the tutorials and documentation that I have seen so far has been really helpful!

**Comment by xorsuyash** [comment]
Hi @ashwinexe when i am trying to run ivy.pad function in the cloned repo for octernship i am getting the following error :
![Screenshot from 2023-08-09 18-56-43](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/98162846/bc525303-d79b-4e4f-bd84-ac614cd0a5be)
This is the problem with only octernship repo i have tried this with main ivy repo and pad function is working fine :- 
![Screenshot from 2023-08-09 19-03-34](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/98162846/03e41dec-c666-405f-bdec-a58b387e97be)

How should i resolve it ?

**Reply by ManoBharathi93** [reply]
I am also facing same problem.. 

TASK: 1 paddle.vision.transforms.vflip
IVY MAIN REPO:
Test run: local environment
![image](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/88357044/3846d7a9-09fe-4b1f-a465-88cee51a0c16)
PR:https://github.com/unifyai/ivy/pull/21419
<img width="573" alt="image" src="https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/88357044/e0e7df76-01ba-4739-932a-8c37e601fc9b">





------------------------------------------------------------------------------------------------------------------------------------------------------
TASK: 2  paddle.take_along_axis
IVY_MAIN_REPO:
Test run : local environment
![image](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/88357044/23ec8ba2-e606-4b3e-9e25-e8d052fe421e)

PR: https://github.com/unifyai/ivy/pull/21526
<img width="584" alt="image" src="https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/88357044/33346554-81c1-4d05-a211-c3e8a590d697">

I have completed two frontend functions in ivy main repo  but I couldn't complete single task.

**Reply by ManoBharathi93** [reply]
@ashwinexe  i just want to know that issue is  related to GitHub assigned private repo or ivy main repo  reply as soon as possible. (Deadline 🥲)
- In ivy's main repo, My PR got's merged

**Reply by AlexandraPosta** [reply]
Hello! Please check this: https://github.com/GitHub-Octernships/Ivy-Octernships-ML/discussions/1#discussioncomment-6708643

**Comment by rohinish404** [comment]
'tuple' object has no attribute 'values'
I'm getting this error while running tests. Also its saying that no such file exists when i try to run the tests using pytest command.
-python3 -m pytest <test_file>::<test_function>
I searched abt this in the discord and it says that there was a testing pipeline issue. Is there any way to solve this error?

**Reply by AlexandraPosta** [reply]
Hello! This may be because the Octernship template is one month old and does not contain all of the latest changes from ivy. Please check this: https://github.com/GitHub-Octernships/Ivy-Octernships-ML/discussions/1#discussioncomment-6708643

**Comment by OkeyAmy** [comment]
Please am finding it difficult to find the private repository that was created by GitHub 

![Uploading IMG_20230810_130239.jpg…]()

**Reply by sahil-sagwekar2652** [reply]
Hello @OkeyAmy , this should be the one you are looking for - https://github.com/GitHub-Octernships/ivy-octernships-ml-OkeyAmy

**Reply by OkeyAmy** [reply]
Thank you

**Comment by ManoBharathi93** [comment]
After Applying to GitHub octernship Follow this steps 
Step : 1 Login to GitHub
Step: 2 Click on the profile that on top right corner
<img width="305" alt="image" src="https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/88357044/be2bad08-7827-4752-96fb-35934512d67d">
Step:3 Now you can see a list, click on your organization
<img width="195" alt="image" src="https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/88357044/ac99bafb-8fbd-4652-9ec2-15f196a3ad22">
Step: 4 You can see like this click on it 
<img width="425" alt="image" src="https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/88357044/cc386b5a-66b0-40c8-8f29-69056592247f">
Step 5: You are inside gitHub organization Click on the private repo and clone it and work on it
<img width="923" alt="image" src="https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/88357044/db7abdf5-3d26-409b-8a5e-4a98052fdbe4">

**Comment by ManoBharathi93** [comment]
I have tested some of the function from different frameworks but it's end up getting errors in [ivy-octernships-ml-ManoBharathi93](https://github.com/GitHub-Octernships/ivy-octernships-ml-ManoBharathi93)
but it works well in ivy main repo:
[octernship.pdf](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/files/12319341/octernship.pdf)
you can check this pdf for the error logs. @ashwinexe

**Reply by xorsuyash** [reply]
@ManoBharathi93 i am getting same errors  .

**Reply by AlexandraPosta** [reply]
Hello! We are looking into this. The main issue is that the version of ivy that is currently used for Octernship (one month old) has some integration errors with our current automated tests. That is the reason why the implementation runs as expected in the ivy main repo, but not on the Octernship side.

Before we update the ivy template used by Octernship, I would suggest to create a new branch (maybe call it "test"), set its remote to unifyai/ivy and pull the latest changed from ivy. In this way, you can run your function tests inside the new branch. When all the tests pass, include your function and test routine implementation in master and create a PR from master (again, make sure that ONLY the required functions are changed in master, 2 files). In this way, we can keep the Octernship version intact and still run tests in a different branch.

**Reply by goyalpramod** [reply]
@AlexandraPosta What about the people who have already merged their branch?

**Reply by ManoBharathi93** [reply]
@AlexandraPosta  I have completed two task in paddle frontend which was already merged in master branch of ivy repo and as @goyalpramod  said I have merged my PR's into my private ivy repo assigned by github  and delete the branches.. In my case what should I do ?
Kindly note This two task are completed for github octernship purpose only. Two task are related to paddle frontend only and task completed after i have applied for the octernship.
Task 1: https://github.com/unifyai/ivy/pull/21419
Task 2: https://github.com/unifyai/ivy/pull/21526

**Reply by ManoBharathi93** [reply]
@AlexandraPosta  When we pull changes from remote ivy repo it gives following error: `fatal: refusing to merge unrelated histories`

**Reply by ManoBharathi93** [reply]
https://github.com/GitHub-Octernships/Ivy-Octernships-ML/discussions/9 kindly look into this...

**Reply by sahil-sagwekar2652** [reply]
@ManoBharathi93 create a new branch and then set unify-ai/ivy as upstream and then pull the latest version into the new branch.

**Reply by SumantxD** [reply]
@AlexandraPosta Will my application be considered if all my tests are failing in the private repo?
but all the tests pass when i have made the PR in the main ivy repo https://github.com/unifyai/ivy/pull/21737

**Reply by ManoBharathi93** [reply]
@sahil-sagwekar2652 This steps pull the changes to my new branches but didn't solve the issue.

**Reply by AlexandraPosta** [reply]
Hello @ManoBharathi93, given that you have functions merged in ivy already, you can go through our normal application cycle as well. Unfortunatly, PR merges in the main ivy repository will not be accounted for the Octernship application.

**Reply by AlexandraPosta** [reply]
Hello @SumantxD, please check [this](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/discussions/15)

**Reply by ManoBharathi93** [reply]
@AlexandraPosta but I have done that task for as part of octernship application. Because of private repository issue, I have merged both on private and public to make it as proof for my submission that I have finished the task.  The task are completed within alloted time period for submission. 🥺

**Comment by goyalpramod** [comment]
I have merged the pr in my private octernship repo. How do I know if the submission has been made?

**Reply by ManoBharathi93** [reply]
Did all the test cases are passed in private repo ? @goyalpramod

**Reply by goyalpramod** [reply]
No they did not, But I read somewhere that we can ignore them. I am not sure where, I actually saw your question earlier and wanted to reply by quoting the exact post where I read we can skip the tests. But I couldn't find it. I believe ashwin can give us some clarity on this.

**Reply by ManoBharathi93** [reply]
Oh okay.. @ashwinexe only person to give clarity on this issues ! .

**Comment by jatinS-dev** [comment]
@ashwinexe 
I made the PR, do I have to merge it in the main too? or someone who will get assigned(as in the public PR) will do that?

**Reply by ManoBharathi93** [reply]
https://github.com/GitHub-Octernships/Ivy-Octernships-ML/discussions/1#discussioncomment-6660906

**Comment by rohinish404** [comment]
I made the pr in my private repo but most of the tests are failing..is there any to know where the tests are failing so that i can make changes? or should i merge it directly?

**Reply by rogn02** [reply]
I have the same issue, it says that I have a payment issue

**Reply by AlexandraPosta** [reply]
Please check [this](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/discussions/15)

**Comment by AKACHI-4** [comment]
hey all, im stucked in some testcases and working on them but just I saw that deadlines has been extended to 23 august can you please clarify ? any one of u ? 

Like is it official or not ? 

@ashwinexe 
@AlexandraPosta

**Reply by sahil-sagwekar2652** [reply]
Where did you see it? I believe it's still 16th Aug

**Reply by AKACHI-4** [reply]
in octernship page there is mentioned that you can apply on the octernship till 23rd of august 

here's the ss 

![image](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/99159580/5e97e57e-4a64-4e17-ad8b-88681cbf8776)

**Reply by AKACHI-4** [reply]
Idk ... whether it's a mistake or still today is last day of submission

**Reply by ashwinexe** [reply]
You are correct the deadline has been extended: [Announcement Post](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/discussions/19#discussion-5525836)

**Reply by Sanchay-T** [reply]
@ashwinexe I am having some trouble and doubts , I have implemented the frontend function and its working properly , but I am not able to run the test , and I went through the docs and also the ivy octernship stream , can you help me?

**Reply by ManoBharathi93** [reply]
Hi @ashwinexe , I have doubt regarding submission. As you know earlier the private repo has an issue, so I worked on main ivy repo and completed the task and use the same code for my private repo submission for octernship, ( task are submitted in the period of octernship. Will you count my submission! ?

**Reply by AKACHI-4** [reply]
Continuing the question .... if task are submitted in period of octernship but merged after the period is it okay ?? .... like our task will be considered submitted or not ??

**Reply by ManoBharathi93** [reply]
Yes @AKACHI-4  ,   my task are merged to main branch of main ivy repo and as part of octernship to submit the task I have merged in private repo also @ashwinexe pls could you respond to this question.

**Comment by rohinish404** [comment]
I saw that the deadline has been extended..but i merged the pr i create in my private repo without testing as there were some issues running the tests and the deadline was close. Can i delete this one and generate a new private repo? And contribute again?

**Comment by xorsuyash** [comment]
@ashwinexe i have completed my ivy octernship task and i tried testing it on ivy main repo and all the tests are passing and for obvious reasons tests are not working on ivy octernship repo but anyways i commited all the changes on  my master branch and pushed it directly to my private octernship repo , i have tried creating a branch test and also set its remote as ivy main repo but i am unable to pull and merge recent changes from ivy main repo to octernship repo so my tests are still not working in octernship repo what should i do ?

**Reply by AlexandraPosta** [reply]
Creating a branch and pulling main is one way to run your tests inside your Octernship private repo. Given that you tested your implementation inside ivy main repo and all tests passed means that your solution should technically be correct. In this case, you can avoid generating that additional branch.

**Comment by Shivam7-1** [comment]
Hii @ashwinexe I am new to Octernships program 
I am unable to apply for IVY Octernships Program getting Error Like this 
" Oh oh, looks like you need to contribute a bit more to the GitHub community before you’re eligible for an Octernship. Check out our guide on how to contribute to open-source projects to learn how you can make a meaningful impact! Keep up the great work and you’ll be a perfect fit in no time! "
Can Anyone Tell me how much Should I Contribute to GitHub for Become Eligible For Octernship Program ?

**Reply by sahil-sagwekar2652** [reply]
The requirement is contributions in every month, it might be that you haven't had a green square in this month. Try doing that, I've seen this work for someone else.

**Reply by Shivam7-1** [reply]
Hii @sahil-sagwekar2652 I tried this also but it didn't worked for me same issue I am getting
Is there any other solution I can try ?

**Reply by sahil-sagwekar2652** [reply]
Try looking for answers in Octernship discussions, I think quite a few people were facing this issue - https://github.com/orgs/community/discussions/categories/github-education?discussions_q=is%3Aopen+category%3A%22GitHub+Education%22+octernship

**Comment by OkeyAmy** [comment]
I made changes in my new branch and created a pull request but the pull request was closed automatically. Please how can I fix it so that I can merge it to my main branch?

**Reply by AlexandraPosta** [reply]
Are there any errors on GitHub? 

Do you happen to have more GitHub accounts and logged in, by mistake, in a different one? (as in are you using the account that was used for the octernship application)

**Comment by xorsuyash** [comment]
Hello everyone , initially i was trying to impliment paddle.pad function for my octernship task and working on this issue for a while i have realized that the way paddle.pad function implemented in paddle is insconsistent with ivy test_frontend_function , to support my inference i have mentioned all inconsistencies in the following doc :
https://docs.google.com/document/d/1fuZ6BOfqqukmQWeEE2x4lZ-pVtM6IbudAF56G2k_uAA/edit

please have a look at this . 
to be on safe side i have implimented flatten function and its test all the test are passing .

**Reply by AlexandraPosta** [reply]
Thank you for your contribution and comments, we will have a look at the document and take it into consideration. Implementing a new function was a good call, well done for completing that! :)

**Reply by goyalpramod** [reply]
Awesome explanation @xorsuyash I will also try working on this.

**Reply by goyalpramod** [reply]
Could you please elaborate what you meant by 

```
...but while in this case numpy arguments works but in some other cases paddle.tensor arguments are providing the best result . so tests keep failing no matter how i control the arguments for frontend_ret..
```

What do you mean by they give the best results? do they pass a majority of the edge cases? 

Also

```
....when we are using paddle.tensor then for some reason paddle allows dim size of range [3,4,5] and when we use it for sampling it branches out into further inconsistencies . 
```

What do you mean by further inconsistencies here? and what is the issue if paddle allows for more dim size than in the tests? 

My questions may seem naive, but these will help me get a better understanding of the problem.

**Reply by xorsuyash** [reply]
@goyalpramod answer to to your first question is when we try to take two different dtype for same tensor and same padding then we get different result by paddle.pad as demonstrated in my example in the doc ,  now our hypothesis module spits out this specific example where my tests was failing in this specific case np.array was working fine and i tweaked things accordingly, but when i did that i got another error message where numpy array input to paddle.pad function was not producing array of same shape as out pad implementation and in that case i tried paddle.tensor and it worked :smiling_face_with_tear: . So paddle.pad behaves very weirdly in different scenarios , you can refer to paddle code base , link is in the doc .

**Reply by xorsuyash** [reply]
Now proceeding to your second question , when we are considering paddle.tensor then while sampling array during test we can only specify dim size of 3,4 and 5 .
Now lets understand the reason behind it , when use paddle.tensor as a input in paddle.pad it calls functional_tensor.pad and functional_tensor.pad pad again calls paddle.nn.functional.pad and then computes the output , 

![Screenshot from 2023-08-23 12-57-05](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/98162846/4f249e03-f1e0-4a6b-954d-0c32e6c47deb)

Now one more thing we need to take in consideration while using paddle.tensor is data format which is not the input for ivy.pad and we cannot sample it while generating arrays because it is different for different dim size and this will also show error . 

![Screenshot from 2023-08-23 12-57-36](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/98162846/8ecdbcbc-ad40-405a-acff-f8c5d51c0348)

Here F is paddle.nn.functional .

**Reply by goyalpramod** [reply]
Oh, Thank you for taking the time and giving such a nice explanation @xorsuyash. It is very weird that it will work differently on different dtypes, I have never encountered such an issue. And I was not aware of the fact that ivy.pad is different for different dim size.

**Comment by Ambro19** [comment]
Hey @ashwinexe, Is this link https://github.com/unifyai/ivy/issues/15037 part of the Octernships assignment too? I am working on gather_nd function. Because there is no more "open task" in this link https://github.com/unifyai/ivy/issues/19178

**Reply by AlexandraPosta** [reply]
Hello! 

Unfortunately, it is not part from the Octernship assignment. 

As far as I can tell, these issues have not been referenced: 
- https://github.com/unifyai/ivy/issues/19168
- https://github.com/unifyai/ivy/issues/19169
- https://github.com/unifyai/ivy/issues/19170
- https://github.com/unifyai/ivy/issues/19171
- https://github.com/unifyai/ivy/issues/19172

**Reply by Sanchay-T** [reply]
@AlexandraPosta I am doing tolist , I have also pushed it and is under review with Vaibhav thakkar. 
See this link : [https://github.com/unifyai/ivy/pull/21303](url)

**Reply by Sanchay-T** [reply]
My 3 test cases are successful , but for tensorflow and pytorch its not working. I am not able to understand whats wrong.

**Reply by AlexandraPosta** [reply]
I would suggest checking the implementation of the function in other frontends as well. Alternatively, you can add a question about the errors that you encounter in our [Discord](https://discord.com/channels/799879767196958751/933380219832762430).

**Reply by Sanchay-T** [reply]
No one is replying to me there , i tried asking and taggin @IVY also

**Comment by samthakur587** [comment]
hey @ashwinexe ,
i am sam. and i am actively contributing to the ivy from last 3 months. I had 11 PR merged in ivy and now i applied for github octernship 2 days ago but when i am try to PR on my private octernship repo it automatically getting close saying that you are not allowed to make PR in this repo. can you please tell what should i do make PR. is that somthing that i am missing to complete before making the PR.

![Screenshot 2023-08-21 202227](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/83540902/ffbbb377-0e0b-4095-b123-2b8e6d6918c3)

**Reply by ashwinexe** [reply]
Hi there, I have deleted the file causing this issue. You can reopen the PRs now :)

**Reply by Sanchay-T** [reply]
I have also completed the part , but some of the test cases are failing how can I push the same into my octernship repo

**Reply by samthakur587** [reply]
thanks @ashwinexe

**Comment by samthakur587** [comment]
I had completed the 7 task from list of task in local but i am not able to make PR 😢 . @ashwinexe can you please check so I can PR before the deadline.

**Comment by Sai-Suraj-27** [comment]


**Reply by ashwinexe** [reply]
> @ashwinexe sir, When can we expect the results? Will there be another round after reviewing our private repos? If yes, how will we be notified? Also, Will we receive any mail about the next rounds?
> 
> Please clarify on these, if you can. Thank you.

You will receive email about your application status once results are out.

**Comment by neeveshjain** [comment]
can anyone help me to understand ivy octernships better ?

**Reply by AKACHI-4** [reply]
Hey @neeveshjain I can help but not now... on weekends so u can connect with me and ping me on weekends ..

**Reply by neeveshjain** [reply]
yes we can connect on weekend , i will ping you.

**Comment by Sanchay-T** [comment]
@ashwinexe My 1 test was not passing , will I be selected? I tried asking in discord and in the PR , but no one helped. I did what I could be going thorough the documentation and the youtube webinar.

**Reply by ashwinexe** [reply]
You can mention that in your PR. Explaining the problem and providing the solutions you tried helps maintainers understand your approach so they can support you. 

Please merge the PR if you haven't already since it's past deadline.

**Reply by Sanchay-T** [reply]
I have created a pr in my repo that was given , what extra should I mention over there

**Reply by Sanchay-T** [reply]
@ashwinexe I had created a PR before deadline and did not merge it , will it count. I have not made any changes, its a single PR

**Comment by Sai-Suraj-27** [comment]
@ashwinexe sir, when can we expect the results? are they out already? if not can you give an approximate date?

**Reply by ashwinexe** [reply]
We will post an announcement discussion once results are you. You will also receive an email with updates on your application status.

**Comment by xorsuyash** [comment]
@ashwinexe
@AlexandraPosta 
 whats the status of octernship result ?, please let us know
